pkg/detectors/jdbc/postgres.go

```diff
@@ -59,41 +59,60 @@ func joinKeyValues(m map[string]string, sep string) string {
 
 func parsePostgres(subname string) (jdbc, error) {
 	// expected form: [subprotocol:]//[user:password@]HOST[/DB][?key=val[&key=val]]
-	hostAndDB, paramString, _ := strings.Cut(subname, "?")
-	if !strings.HasPrefix(hostAndDB, "//") {
+
+	if !strings.HasPrefix(subname, "//") {
 		return nil, errors.New("expected host to start with //")
 	}
-	userPassAndHostAndDB := strings.TrimPrefix(hostAndDB, "//")
-	userPass, hostAndDB, found := strings.Cut(userPassAndHostAndDB, "@")
-	var user, pass string
-	if found {
-		user, pass, _ = strings.Cut(userPass, ":")
-	} else {
-		hostAndDB = userPass
+
+	u, err := url.Parse(subname)
+	if err != nil {
+		return nil, err
 	}
-	host, database, found := strings.Cut(hostAndDB, "/")
-	if !found {
-		return nil, errors.New("expected host and database to be separated by /")
+
+	dbName := strings.TrimPrefix(u.Path, "/")
+	if dbName == "" {
+		dbName = "postgres"
 	}
 
 	params := map[string]string{
-		"host":   host,
-		"dbname": database,
+		"host":            u.Host,
+		"dbname":          dbName,
+		"connect_timeout": "5",
 	}
-	if len(user) > 0 {
-		params["user"] = user
-	}
-	if len(pass) > 0 {
-		params["password"] = pass
+
+	if u.User != nil {
+		params["user"] = u.User.Username()
+		pass, set := u.User.Password()
+		if set {
+			params["password"] = pass
+		}
 	}
-	for _, param := range strings.Split(paramString, "&") {
-		key, val, _ := strings.Cut(param, "=")
-		params[key] = val
+
+	for k, v := range u.Query() {
+		switch k {
+		case "sslmode":
+			if isAlphabet(v[0]) {
+				params[k] = v[0]
+			}
+		case "user":
+			params[k] = v[0]
+		case "password":
+			params[k] = v[0]
+		}
 	}
```

Comment:
Is it guaranteed that `len(v) >= 1`?

Reply:
Also, why does `sslmode` require an alphabetic value?

Reply:
+1 Should we check `sslmode` against a list of expected values?

Reply:
Also, can't it also have a `-`? eg `verify-ca`

Reply:
Added length check and check against expected values.